src/huggingface_hub/utils/_errors.py

```diff
@@ -357,6 +357,17 @@ def hf_raise_for_status(response: Response, endpoint_name: Optional[str] = None)
             )
             raise BadRequestError(message, response=response) from e
 
+        elif response.status_code == 403:
+            message = (
+                f"{response.status_code} Client Error." + "\n\n" + f"Bad request for {endpoint_name} endpoint:"
+                if endpoint_name is not None
+                else "Bad request:"
+                + "\nPlease make sure you specified the correct `repo_id` and `repo_type`."
+                + "\nIf you are trying to create or update content,"
+                + "make sure you have a token with the `write` role."
+            )
+            raise BadRequestError(message, response=response) from e
```

Comment:
Hey @bmuskalla, thanks for the very clean and concise PR! :rocket:  

Everything looks good to me except that in case of a `HTTP 403` I would raise a generic `HfHubHTTPError` instead of a `BadRequestError`. A Bad Request error might be misinterpreted as a HTTP 400 which is not the case here. Regarding the error message I would make it simpler:
```py
message = f"{e}\nIf you are trying to create or update content, make sure you have a token with the `write` role."
raise HfHubHTTPError(message, response=response) from e
```
What do you think?

Reply:
Point taken on the confusing nature of using `BadRequestError`, changed to `HfHubHTTPError`. 
For the message, I've played a bit more with a real error and ended up using Forbidden+server-side message + the hint.

Example:
```
Traceback (most recent call last):
  File "/Users/bmuskalla/git/huggingface_hub/play.py", line 5, in <module>
    create_repo(repo_id="bmuskalla/super-cool-model", token=HF_RO_TOKEN)
  File "/Users/bmuskalla/git/huggingface_hub/src/huggingface_hub/utils/_validators.py", line 118, in _inner_fn
    return fn(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^
  File "/Users/bmuskalla/git/huggingface_hub/src/huggingface_hub/hf_api.py", line 3352, in create_repo
    hf_raise_for_status(r)
  File "/Users/bmuskalla/git/huggingface_hub/src/huggingface_hub/utils/_errors.py", line 367, in hf_raise_for_status
    raise HfHubHTTPError(message, response=response) from e
huggingface_hub.utils._errors.HfHubHTTPError:  (Request ID: Root=1-65df9500-64c1f6dd124322f75dae5c8f;b7c328b6-c3fa-47f6-830e-a488fca19cdd)

403 Forbidden: You don't have the rights to create a model under this namespace.
Cannot access content at: https://huggingface.co/api/repos/create.
If you are trying to create or update content,make sure you have a token with the `write` role.
```

I found this the most readable but it's your call in the end.